### PR TITLE
Initial Project Import

### PR DESCRIPTION
**Pull request by twz123**
Import the initial project structure.

### CODE REVIEW

File: src/main/java/org/zalando/undertaking/inject/guice/PrivateHttpExchangeModule.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.google.inject.Exposed;
+import com.google.inject.PrivateBinder;
+import com.google.inject.PrivateModule;
+import com.google.inject.Provides;
+
+import io.undertow.server.HttpServerExchange;
+
+import io.undertow.util.HeaderMap;
+
+// here go bindings that need the HttpServerExchange instance
```
Comment (mschwachenbe-zalando) [inline src/main/java/org/zalando/undertaking/inject/guice/PrivateHttpExchangeModule.java]: Could you change this comment into a Javadoc?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  - oraclejdk8
+
+sudo: false
+
+cache:
+  directories:
+    - $HOME/.m2
+
+script:
+  - ./mvnw clean verify
```
Comment (jans510) [inline .travis.yml]: As travis has successfully built it I assume mvnw is correct (and not mvn).

Comment (jans510) [inline .travis.yml]: Got it! Maven wrapper :smile:

File: README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Undertaking
+===========
+
+A toolbox for Undertow/Rx/Guice.
```
Comment (jans510) [inline README.md]: That's all about it, nothing more to say :smile:

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        <!-- Reactive and friends -->
+        <dependency>
+            <groupId>io.reactivex</groupId>
+            <artifactId>rxjava</artifactId>
+            <version>1.1.9</version>
+        </dependency>
+        <dependency>
+            <groupId>com.netflix.hystrix</groupId>
+            <artifactId>hystrix-core</artifactId>
+            <version>1.4.26</version>
```
Comment (jans510) [inline pom.xml]: You have defined the version as parameter.

**Comment by twz123** [comment]
:+1:

**Comment by mschwachenbe-zalando** [comment]
:+1:

File: src/main/java/org/zalando/undertaking/inject/HttpExchangeScope.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @param   wrapped {@code HttpHandler} that shall be wrapped
+     *
+     * @return  a wrapping {@code HttpHandler} that scopes the execution of the wrapped {@code HttpHandler} to the
+     *          actual {@code HttpServerExchange} that gets passed to it
+     *
+     * @throws  NullPointerException  if {@code wrapped} is {@code null}
+     */
+    default HttpHandler scoped(final HttpHandler wrapped) {
+        checkNotNull(wrapped);
```
Comment (jans510) [inline src/main/java/org/zalando/undertaking/inject/HttpExchangeScope.java]: any reason why you don't use requireNonNull?

File: src/main/java/org/zalando/undertaking/logbook/UndertowHttpMessage.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+import io.undertow.server.HttpServerExchange;
+
+import io.undertow.util.HeaderMap;
+import io.undertow.util.HeaderValues;
+import io.undertow.util.Headers;
+
+public abstract class UndertowHttpMessage implements HttpMessage {
+
+    private static final byte[] EMPTY = new byte[0];
```
Comment (jans510) [inline src/main/java/org/zalando/undertaking/logbook/UndertowHttpMessage.java]: That's weird :smile:

**Comment by twz123** [comment]
:+1:

**Comment by twz123** [comment]
:+1:

**Comment by jans510** [comment]
:+1:

File: pom.xml (hunk 4 above)
Comment (twz123) [inline pom.xml]: Yeah, right. Removed the parameter.

**Pull request merged**